skimage/metrics/set_metrics.py

```diff
@@ -49,3 +52,65 @@ def hausdorff_distance(image0, image1):
 
     return max(max(cKDTree(a_points).query(b_points, k=1)[0]),
                max(cKDTree(b_points).query(a_points, k=1)[0]))
+
+
+def hausdorff_pair(image0, image1):
+    """Returns pair of points that are Hausdorff distance apart between nonzero
+    elements of given images.
+
+    The Hausdorff distance [1]_ is the maximum distance between any point on
+    ``image0`` and its nearest point on ``image1``, and vice-versa.
+
+    Parameters
+    ----------
+    image0, image1 : ndarray
+        Arrays where ``True`` represents a point that is included in a
+        set of points. Both arrays must have the same shape.
+
+    Returns
+    -------
+    point_a, point_b : array
+        A pair of points that have Hausdorff distance between them.
+
+    References
+    ----------
+    .. [1] http://en.wikipedia.org/wiki/Hausdorff_distance
+
+    Examples
+    --------
+    >>> points_a = (3, 0)
+    >>> points_b = (6, 0)
+    >>> shape = (7, 1)
+    >>> image_a = np.zeros(shape, dtype=bool)
+    >>> image_b = np.zeros(shape, dtype=bool)
+    >>> image_a[points_a] = True
+    >>> image_b[points_b] = True
+    >>> hausdorff_pair(image_a, image_b)
+    (array([3, 0]), array([6, 0]))
+
```

Comment:
Can't we demonstrate here the relation between Hausdorff distance and Hausdorff pair:
```python
Hausdorff distance is the Euclidean distance between Hausdorff pair:

>>> from scipy.spatial import distance
>>> d = distance.euclidean(point_a, point_b))
>>> hausdorff_distance(image_a, image_b) == d
True
```
?

Reply:
While this works, I feel like the python docstring for a method should show the method's functionality and not rely on anything outside the method itself ie `hausdorff_distance`. Maybe we can put this check in the unit tests?

Reply:
Sure. Please include it in `skimage/metrics/tests/test_set_metrics.py`.

Reply:
Done! Note that I had to change the names of the `distance` variables so there's less ambiguity between `scipy.spatial.distance` and the distances between the points.